test/lib/jdk/test/lib/util/ForceGC.java

```diff
@@ -100,6 +99,7 @@ public static boolean waitFor(BooleanSupplier booleanSupplier, long timeout) {
                 // ignore, the loop will try again
             }
         }
+        Reference.reachabilityFence(ref);
```

Comment:
I think everything from the creation of ref to the line above needs to enclosed in a try-statement, with the finally-clause including RF(ref).

Reply:
Arguably the same might apply to the other call to reachability fence.